protocol/scripts/deploys/mainnet/27-deploy-payments-factory.ts

```diff
@@ -0,0 +1,56 @@
+import '@nomiclabs/hardhat-ethers';
+import { ethers, network } from 'hardhat';
+import { TempleTeamPaymentsFactory__factory } from '../../../typechain';
+import {
+  DeployedContracts,
+  DEPLOYED_CONTRACTS,
+  expectAddressWithPrivateKey,
+  mine,
+} from '../helpers';
+
```

Comment:
add a new file for the deploy script for the impelentation contract if you decide to deploy the implementation contract separately like i suggested.